contracts/lib/OrderValidator.sol

```diff
@@ -290,6 +310,189 @@ contract OrderValidator is Executor, ZoneInteraction {
         return (orderHash, numerator, denominator);
     }
 
+    function _getGeneratedOrder(
+        OrderParameters memory orderParameters,
+        bytes memory context
+    ) internal returns (bytes32 orderHash) {
+        // TODO: reuse an existing memory region or relocate this functionality
+        (
+            SpentItem[] memory originalOfferItems,
+            SpentItem[] memory originalConsiderationItems
+        ) = _convertToSpent(
+                orderParameters.offer,
+                orderParameters.consideration
+            );
+
+        // TODO: allow for skipping in cases where the call reverts
+        (
+            SpentItem[] memory offer,
+            ReceivedItem[] memory consideration
+        ) = ContractOffererInterface(orderParameters.offerer).generateOrder(
+                originalOfferItems,
+                originalConsiderationItems,
+                context
+            );
+
+        // Explicitly specified offer items cannot be removed.
+        if (orderParameters.offer.length > offer.length) {
+            _revertNoSpecifiedOrdersAvailable(); // TODO: replace w/ better err
+        }
+
+        // Loop through each offer and ensure at least as much on returned offer
+        for (uint256 i = 0; i < orderParameters.offer.length; ++i) {
+            OfferItem memory originalOffer = orderParameters.offer[i];
+            SpentItem memory newOffer = offer[i];
+
+            if (
+                originalOffer.startAmount != originalOffer.endAmount ||
+                originalOffer.endAmount > newOffer.amount ||
+                originalOffer.itemType != newOffer.itemType ||
+                originalOffer.token != newOffer.token ||
+                originalOffer.identifierOrCriteria != newOffer.identifier
+            ) {
+                _revertNoSpecifiedOrdersAvailable(); // TODO: replace
+            }
+
+            originalOffer.startAmount = newOffer.amount;
+            originalOffer.endAmount = newOffer.amount;
+        }
+
+        // add new offer items if there are more than original
+        for (
+            uint256 i = orderParameters.offer.length - 1;
+            i < offer.length;
+            ++i
+        ) {
+            OfferItem memory originalOffer = orderParameters.offer[i];
```

Comment:
Is assumption here that `orderParameters.offer` will have free memory region to write extra offer items to?
We'd have to know the upper limit of number of OfferItems that will be returned by the contract to assign a safe region of memory, right? 🤔